src/main/java/neuefische/capstone/backend/security/service/CredentialService.java

```diff
@@ -0,0 +1,39 @@
+package neuefische.capstone.backend.security.service;
+
+import neuefische.capstone.backend.security.userCredentialModel.Credential;
+import neuefische.capstone.backend.security.storage.CredentialRepo;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Component;
+
+import java.util.Optional;
+
+
+@Component
+public class CredentialService implements UserDetailsService {
+    private final CredentialRepo repo;
+
+    @Autowired
+    public CredentialService(CredentialRepo repo) {
+        this.repo = repo;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String username)
+            throws UsernameNotFoundException {
+        Optional<Credential> maybeCredential = repo.findByUsername(username);
+        return maybeCredential
```

Comment:
```suggestion
        return repo.findByUsername(username)
```
no need for the other line, orElseThrow takes care of that